pkg/cmd/spks.go

```diff
@@ -0,0 +1,215 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"github.com/urfave/cli/v2"
+	"github.com/vshn/billing-collector-cloudservices/pkg/log"
+	"github.com/vshn/billing-collector-cloudservices/pkg/odoo"
+)
+
+func SpksCMD() *cli.Command {
+	var (
+		prometheusQueryArr = [4]string{
+			"count(max_over_time(crossplane_resource_info{kind=\"compositemariadbinstances\", service_level=\"standard\"}[1d:1d]))",
+			"count(max_over_time(crossplane_resource_info{kind=\"compositemariadbinstances\", service_level=\"premium\"}[1d:1d]))",
+			"count(max_over_time(crossplane_resource_info{kind=\"compositeredisinstances\", service_level=\"standard\"}[1d:1d]))",
+			"count(max_over_time(crossplane_resource_info{kind=\"compositeredisinstances\", service_level=\"premium\"}[1d:1d]))",
+		}
+		odooURL           string
+		odooOauthTokenURL string
+		odooClientId      string
+		odooClientSecret  string
+		salesOrder        string
+		prometheusURL     string
+		UnitID            string
+	)
+
+	return &cli.Command{
+		Name:   "spks",
+		Usage:  "Collect metrics from spks",
+		Before: addCommandName,
+		Flags: []cli.Flag{
+			&cli.StringFlag{Name: "odoo-url", Usage: "URL of the Odoo Metered Billing API",
+				EnvVars: []string{"ODOO_URL"}, Destination: &odooURL, Value: "https://preprod.central.vshn.ch/api/v2/product_usage_report_POST"},
+			&cli.StringFlag{Name: "odoo-oauth-token-url", Usage: "Oauth Token URL to authenticate with Odoo metered billing API",
+				EnvVars: []string{"ODOO_OAUTH_TOKEN_URL"}, Destination: &odooOauthTokenURL, Required: true, DefaultText: defaultTextForRequiredFlags},
+			&cli.StringFlag{Name: "odoo-oauth-client-id", Usage: "Client ID of the oauth client to interact with Odoo metered billing API",
+				EnvVars: []string{"ODOO_OAUTH_CLIENT_ID"}, Destination: &odooClientId, Required: true, DefaultText: defaultTextForRequiredFlags},
+			&cli.StringFlag{Name: "odoo-oauth-client-secret", Usage: "Client secret of the oauth client to interact with Odoo metered billing API",
+				EnvVars: []string{"ODOO_OAUTH_CLIENT_SECRET"}, Destination: &odooClientSecret, Required: true, DefaultText: defaultTextForRequiredFlags},
+			&cli.StringFlag{Name: "sales-order", Usage: "Sales order for APPUiO Managed clusters",
+				EnvVars: []string{"SALES_ORDER"}, Destination: &salesOrder, Required: false, DefaultText: defaultTextForOptionalFlags, Value: "S10121"},
+			&cli.StringFlag{Name: "prometheus-url", Usage: "URL of the Prometheus API",
+				EnvVars: []string{"PROMETHEUS_URL"}, Destination: &prometheusURL, Required: true, DefaultText: defaultTextForRequiredFlags, Value: "http://prometheus-monitoring-application.monitoring-application.svc.cluster.local:9090"},
+			&cli.StringFlag{Name: "unit-id", Usage: "Unit ID for the consumed units",
```

Comment:
```suggestion
			&cli.StringFlag{Name: "unit-id", Usage: "Metered Billing UoM ID for the consumed units",
```

This is more clear IMHO